proto/proto/strapper.proto

```diff
@@ -10,9 +10,17 @@ message Interface {
 	repeated string ipaddr = 3;
 }
 
+message PublicKeys {
+	string rsa = 1;
+	string dsa = 2;
+	string ecdsa = 3;
+	string ed25519 = 4;
+}
+
 message NodeAdvertisement {
 	string hostname = 1;
 	repeated Interface interfaces = 2;
+	repeated PublicKeys pubkeys = 3;
```

Comment:
This doesn't make sense - a host will only have one set of keys.

Perhaps map<string, string> might be a better type for this?